src/ucp/core/ucp_ep.c

```diff
@@ -688,60 +694,86 @@ int ucp_ep_config_is_equal(const ucp_ep_config_key_t *key1,
-
-    /* We calculate the Rendezvous threshold by finding the message size at which:
-     * AM/RMA rndv's latency is worse than the eager_zcopy
-     * latency by a small percentage (that is set by the user).
-     * Starting this message size (rndv_thresh), rndv may be used.
-     *
-     * The latency function for eager_zcopy is:
-     * [ reg_cost.overhead + size * md_attr->reg_cost.growth +
-     * max(size/bw , size/bcopy_bw) + overhead ]
-     *
-     * The latency function for Active message Rendezvous is:
-     * [ latency + overhead + reg_cost.overhead +
-     * size * md_attr->reg_cost.growth + overhead + latency +
-     * max(size/bw , size/bcopy_bw) + latency + overhead + latency ]
-     *
-     * The latency function for RMA (get_zcopy) Rendezvous is:
-     * [ reg_cost.overhead + size * md_attr->reg_cost.growth + latency + overhead +
-     *   reg_cost.overhead + size * md_attr->reg_cost.growth + overhead + latency +
-     *   size/bw + latency + overhead + latency ]
-     *
-     * Isolating the 'size' yields the rndv_thresh.
-     * The used latency functions for eager_zcopy and rndv are also specified in
-     * the UCX wiki */
-
-    if (md_attr->cap.flags & UCT_MD_FLAG_REG) {
-        md_reg_growth   = md_attr->reg_cost.growth;
-        md_reg_overhead = md_attr->reg_cost.overhead;
-    } else {
-        md_reg_growth   = 0;
-        md_reg_overhead = 0;
+    ucp_context_h context = worker->context;
+    ucp_md_map_t md_map   = 0;
+    ucp_lane_index_t lane;
+    ucp_rsc_index_t rsc_index;
+    ucp_md_index_t md_index;
+    uct_md_attr_t *md_attr;
+    int i;
+
+    for(i = 0; lanes[i] != UCP_NULL_LANE; i++) {
+        lane      = lanes[i];
+        rsc_index = config->key.lanes[lane].rsc_index;
+        md_index  = config->md_index[lane];
+
+        if (!(md_map & UCS_BIT(md_index))) {
+            md_map |= UCS_BIT(md_index);
+            md_attr = &context->tl_mds[md_index].attr;
+            if (md_attr->cap.flags & UCT_MD_FLAG_REG) {
+                params->reg_growth   += md_attr->reg_cost.growth;
+                params->reg_overhead += md_attr->reg_cost.overhead;
+            }
+        }
+        params->bw += worker->ifaces[rsc_index].attr.bandwidth;
     }
+}
 
-    numerator = diff_percent * ((4 * ucp_tl_iface_latency(context, iface_attr)) +
-                (3 * iface_attr->overhead) +
-                (md_reg_overhead * (1 + recv_reg_cost))) -
-                 md_reg_overhead - iface_attr->overhead;
+static size_t ucp_ep_config_calc_rndv_thresh(ucp_worker_t *worker,
+                                             ucp_ep_config_t *config,
+                                             ucp_lane_index_t *eager_lanes,
+                                             ucp_lane_index_t *rndv_lanes,
+                                             int recv_reg_cost)
+{
+    ucp_context_h context        = worker->context;
+    ucp_ep_thresh_params_t eager = {0};
```

Comment:
would be better if ucp_ep_config_calc_thresh_params() is responsible for initializing to 0, so it would be "out" parameter and not "inout"